File: solutions/erlang/leap/1/src/leap.erl
-module(leap).

-export([leap_year/1]).

% is the year is also evenly divisible by 4 and 400? Leap
leap_year(Year) when Year rem 4 == 0, Year rem 400 == 0 ->
  true;

% is the year evenly divisible by 4 and 100? Not leap
leap_year(Year) when Year rem 4 == 0, Year rem 100 == 0 ->
  false;

% is the year evenly divisible by 4 ? Leap
leap_year(Year) when Year rem 4 == 0 ->
  true;

% is the year not evenly divisible by 4 ? Not leap
leap_year(_Year) ->
  false
.

File: solutions/python/leap/1/leap.py
def leap_year(year):
    """A simple function that determines if a year is a leap year."""
    # leap years are divisible by 400
    if year % 400 == 0:
        return True
    # but leap years are not divisible by 100
    elif year % 100 == 0:
        return False
    # but leap years are divisible by 4
    elif year % 4 == 0:
        return True
    # and years not divisible by 4 are not leap years
    else:
        return False
